tests/test_generation.py

```diff
@@ -0,0 +1,178 @@
+from textwrap import dedent
+
+import jax.numpy as jnp
+import pytest
+from attr import dataclass
+from jax import jit, vmap
+from jaxtyping import Array, Int
+from transformers import AutoTokenizer
+from transformers.tokenization_utils import PreTrainedTokenizer
+
+from lalamo.language_model import LanguageModel
+from lalamo.model_import import REPO_TO_MODEL, import_model
+
+
+@dataclass
+class GenerationInput:
+    prompt: str
+    token_ids: Int[Array, " tokens"]
+
+
+@pytest.fixture
+def generation_input(tokenizer: PreTrainedTokenizer) -> GenerationInput:
+    prompt = dedent(
+        """
+        <|im_start|>user
+        hello<|im_end|>
+        <|im_start|>assistant
+        <think>\n\n</think>\n\n
+    """.lstrip(),
+    )
+    tokens = jnp.array(tokenizer.encode(prompt))
+    return GenerationInput(prompt, tokens)
+
+
+@pytest.fixture
+def another_generation_input(tokenizer: PreTrainedTokenizer) -> GenerationInput:
+    prompt = dedent(
+        """
+        <|im_start|>user
+        How are you?<|im_end|>
+        <|im_start|>assistant
+        <think>\n\n</think>\n\n
+    """.lstrip(),
+    )
+    tokens = jnp.array(tokenizer.encode(prompt))
+    return GenerationInput(prompt, tokens)
+
+
+@pytest.fixture
+def language_model() -> LanguageModel:
+    decoder = import_model(REPO_TO_MODEL["Qwen/Qwen3-0.6B"]).model
```

Comment:
Maybe this would be more clear?

```suggestion
    decoder = import_model(REPO_TO_MODEL["Qwen/Qwen3-0.6B"]).decoder
```